### ISSUE TITLE
complex Filter form does not respect colspans

### ISSUE THREAD

**Issue by G3z** [filamentphp/filament#4235]
### Package

filament/tables

### Package Version

v2.16.17

### Laravel Version

v9.31.0

### PHP Version

PHP 8.1.10

### Problem description

When building a fairly complex filter form i'm using colspans to organize filters
```php
public static function table(Table $table): Table
    {
        return $table
            ->columns([
                TextColumn::make('name'),
            ])
            ->filters([
                Filter::make('name')->form([
                    TextInput::make('name'),
                ]),
                Filter::make('advanced')
                ->columnSpan(5)
                ->form([
                    Section::make('advanced')
                    ->columnSpan(5)
                    ->collapsed()
                    ->compact()
                    ->schema([
                        TextInput::make('name'),
                        TextInput::make('name'),
                    ]),
                ]),
            ], Layout::AboveContent);
    }
```

but every filter is always inside a `col-span-1`

![Schermata 2022-09-27 alle 15 12 56](https://user-images.githubusercontent.com/875086/192536133-ae7c60cc-33b4-41e7-be22-adad069b9d48.jpg)

### Expected behavior

![Schermata 2022-09-27 alle 15 13 14](https://user-images.githubusercontent.com/875086/192536151-a0a08367-87de-4346-997f-b9c8dc3b6952.jpg)

### Steps to reproduce

use `colSpan()` on a Filter with `Layout::AboveContent`

### Reproduction repository

https://github.com/G3z/filament-bug/commit/b466894524f8f4b2301f9a03e0de9b87600267c0

**Comment by BorisKM** [comment]
The same problem!

**Comment by G3z** [comment]
I think the problem is that every Filter is wrapped in a `Filament\Forms\Components\Group`
that shadows the Filter's colspan
this is a dump of
```php
SelectFilter::make('test')->columnSpan(5)->options([])
```

notice the groud's colspan and the filter's colspan

![Schermata 2022-09-27 alle 17 47 28](https://user-images.githubusercontent.com/875086/192573666-ad686133-b458-4753-bb36-1497cb22f6c6.jpg)

**Comment by danharrin** [comment]
Hey, I've fixed this in #4422.

`Filter::make()->columnSpan(3)` - this will now make the filter subform span 3 columns WITHIN the main filter form. So if your main filter form has 5 columns, this will span 60%.

`Filter::make()->columns(5)` - this will now make the filter CONTAIN 5 columns within it. So if you have 5 filter `form()` components, they will all be next to each other, taking up 20% of the 60% `columnSpan()` - `12%` for each form component.

In your example:

```php
Filter::make('name')->form([
    TextInput::make('name'),
]),
Filter::make('advanced')
->columnSpan(5)
->form([
    Section::make('advanced')
    ->collapsed()
    ->compact()
    ->schema([
        TextInput::make('name'),
        TextInput::make('name'),
    ]),
]),
```
